Modules/Sources/Yosemite/Tools/POS/POSCatalogSyncCoordinator.swift

```diff
@@ -724,23 +770,44 @@ private extension POSCatalogSyncCoordinator {
         if daysSinceFirstSync > Constants.maxDaysSinceLastOpened {
             // More than 30 days since first sync - must have opened POS recently to remain eligible
             guard let lastOpened = lastOpenedDate else {
-                DDLogInfo("📋 POSCatalogSyncCoordinator: Site \(siteID) ineligible (past 30-day grace period, no recent POS open)")
-                return false
+                return .ineligible(reason: .neverOpenedAndPastFirstSyncGracePeriod(daysSinceFirstSync: daysSinceFirstSync))
             }
 
             let daysSinceLastOpened = Calendar.current.dateComponents([.day], from: lastOpened, to: Date()).day ?? 0
 
             if daysSinceLastOpened <= Constants.maxDaysSinceLastOpened {
-                DDLogInfo("📋 POSCatalogSyncCoordinator: Site \(siteID) eligible (last opened \(daysSinceLastOpened) days ago)")
-                return true
+                return .eligible(reason: .openedRecently(daysSinceOpened: daysSinceLastOpened))
             } else {
-                DDLogInfo("📋 POSCatalogSyncCoordinator: Site \(siteID) ineligible (POS last opened \(daysSinceLastOpened) days ago)")
-                return false
+                return .ineligible(reason: .notOpenedRecently(daysSinceOpened: daysSinceLastOpened))
             }
         } else {
             // Within 30 days of first sync - always eligible (new user grace period)
-            DDLogInfo("📋 POSCatalogSyncCoordinator: Site \(siteID) eligible (within grace period: \(daysSinceFirstSync) days since first sync)")
-            return true
+            return .eligible(reason: .recentFirstSync(daysSinceFirstSync: daysSinceFirstSync))
+        }
+    }
+
+    enum SyncEligibility {
+        case eligible(reason: EligibleReason)
+        case ineligible(reason: IneligibleReason)
+
+        enum EligibleReason {
+            case neverSynced
+            case openedRecently(daysSinceOpened: Int)
+            case recentFirstSync(daysSinceFirstSync: Int)
+        }
+
+        enum IneligibleReason {
+            case notEligibleForLocalCatalog
+            case notOpenedRecently(daysSinceOpened: Int)
+            case neverOpenedAndPastFirstSyncGracePeriod(daysSinceFirstSync: Int)
+
+            var skipReason: String {
+                switch self {
+                    case .notEligibleForLocalCatalog: "not_eligible"
+                    case .notOpenedRecently: "pos_not_opened_30_days"
+                    case .neverOpenedAndPastFirstSyncGracePeriod: "pos_not_opened_30_days"
```

Comment:
Is it ok to use the same `pos_not_opened_30_days` value for both cases `notOpenedRecently` and `neverOpenedAndPastFirstSyncGracePeriod`? I noticed it is expected by the coordinator tests, but raising it just in case 👍